File: apps/desktop/renderer/src/types/terminal.ts
export type TerminalKind =
  | "shell"
  | "agent"
  | "dev_server"
  | "test_runner"
  | "script"
  | "task";

export type TerminalStatus =
  | "starting"
  | "running"
  | "exited"
  | "failed"
  | "killed";

export interface TerminalSessionInfo {
  id: string;
  kind: TerminalKind;
  cwd: string;
  shell: string;
  cols: number;
  rows: number;
  status: TerminalStatus;
  created_at: string;
  last_activity_at: string;
}

export interface CreateTerminalRequest {
  cwd?: string;
  shell?: string;
  cols?: number;
  rows?: number;
  kind?: TerminalKind;
}

export interface TerminalOutputChunk {
  session_id: string;
  seq: number;
  data: string;
  timestamp: string;
}

export interface TerminalStateEvent {
  session_id: string;
  status: string;
}

export interface TerminalExitEvent {
  session_id: string;
  exit_code: number | null;
}

export interface AttachSnapshot {
  session: TerminalSessionInfo;
  from_seq: number;
  latest_seq: number;
  recent_chunks: TerminalOutputChunk[];
}

File: crates/orphix-core/src/protocol/terminal.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum TerminalKind {
    Shell,
    Agent,
    DevServer,
    TestRunner,
    Script,
    Task,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[serde(rename_all = "snake_case")]
pub enum TerminalStatus {
    Starting,
    Running,
    Exited,
    Failed,
    Killed,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateTerminalRequest {
    pub cwd: Option<String>,
    pub shell: Option<String>,
    pub cols: Option<u16>,
    pub rows: Option<u16>,
    pub kind: Option<TerminalKind>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TerminalSessionInfo {
    pub id: String,
    pub kind: TerminalKind,
    pub cwd: String,
    pub shell: String,
    pub cols: u16,
    pub rows: u16,
    pub status: TerminalStatus,
    pub created_at: String,
    pub last_activity_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TerminalOutputChunk {
    pub session_id: String,
    pub seq: u64,
    pub data: String,
    pub timestamp: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ShellInfoDto {
    pub program: String,
    pub args: Vec<String>,
    pub label: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AttachSnapshot {
    pub session: TerminalSessionInfo,
    pub from_seq: u64,
    pub latest_seq: u64,
    pub recent_chunks: Vec<TerminalOutputChunk>,
}
